summoner-cli/src/Summoner/Text.hs

```diff
@@ -22,6 +23,18 @@ packageToModule = tconcatMap headToUpper . T.splitOn "-"
 packageNameValid :: Text -> Bool
 packageNameValid = T.all (\c -> c == '-' || C.isAlphaNum c)
 
+{- | Validate module name. It should be in the following formatTriple
+
+@
+Part1[.PartN]
+@
+-}
+moduleNameValid :: Text -> Bool
+moduleNameValid = all (\s -> s /= "" && validFragment s) . T.split (== '.')
+  where
+    validFragment :: Text -> Bool
+    validFragment s = T.all C.isAlphaNum s && C.isUpper (T.head s)
```

Comment:
I think it should be `T.strip s /= ""` just in case to guard from more invalid inputs... Also, what do you think about moving this this `/= ""` check to the `validFragmet` function so on the top-level this can be just?

```haskell
all isValidFragment . T.split ...
```

Reply:
I thought that the case with spaces should be caught in `C.alphaNum`...

Hmm, interesting idea, can move it there!

Reply:
Hmm, indeed. Makes sense!